README.md

```diff
@@ -1,22 +1,91 @@
 
-# Setup
+# Installation
 
-This codebase assumes Python 3.6. Install the requirements in 
-`requirements-build.txt` before those in `requirements.txt`.
-Anaconda users can install directly with `conda env create -f environment.yml`.
+The easiest way to install the code is to build the Docker image in the `Dockerfile`.
+This will install all necessary binary and Python dependencies.
+You can also pull a Docker image for the latest master commit from
+`humancompatibleai/adversarial_policies:latest`.
+If you want to run outside of Docker (for example, for ease of development), read on.
 
-Then install *one of* `requirements-aprl.txt` or `requirements-modelfree.txt`
-depending on which experiments you need to run (unfortunately these codebases
-depend on different MuJoCo and Gym versions.)
+This codebase uses Python 3.6. The main binary dependencies are MuJoCo (version 1.3.1 for
+`modelfree` and 1.5.0 for `aprl`). You may also need to install some other libraries,
+such as OpenMPI.
 
-Finally, `pip install -e .` to install an editable version of the package.
+There are two different parts of the codebase, `src/aprl` and `src/modelfree` with mutually
+exclusive requirements (due to different Gym and MuJoCo versions). You can create a
+virtual environment for each by running `ci/build_venv.sh` with either `aprl` or `modelfree`.
+Finally, inside the relevant virtual environment run `pip install -e .` to install an editable
+version of the package.
+
+# Reproducing Results
+
+## Training adversarial policies
+
+`modelfree.train` trains a single adversarial policy. `modelfree.multi.train` trains multiple
+adversarial policies, using Ray (see below) for parallelization.
+
+To replicate the results in the paper (there may be slight differences due to randomness not
+captured in the seeding), run `python -m modelfree.multi.train with paper`. To run the
+hyperparameter sweep, run `python -m modelfree.multi.train with hyper`.
+
+You can find results from our training run on s3://adversarial-policies-public/multi_train/paper.
+This includes TensorBoard logs, final model weights, checkpoints, and individual policy configs.
+Run `experiments/pull_public_s3.sh` to sync this and other data to `data/aws-public/`.
+
+## Evaluating adversarial policies
+
+`modelfree.score_agent` evaluates a pair of policies, for example an adversary and a victim.
+It outputs the win rate for each agent and the number of ties. It can also render to the screen
+or produce videos.
+
+We similarly use `modelfree.multi.score` to evaluate multiple pairs of policies in parallel.
+To reproduce all the evaluations used in the paper, run the following bash scripts, which call
+`modelfree.multi.score` internally:
+  - `experiments/modelfree/baselines.sh`: fixed baselines (no adversarial policies).
+  - `experiments/modelfree/attack_transfers.sh <path-to-trained-adversaries>`. To use our
+     pre-trained policies, use the path `data/aws-public/multi_train/paper/20190429_011349`.
```

Comment:
Probably obvious, but maybe worth explicitly clarifying "use the path <> after syncing against S3" .

Reply:
added